core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -220,11 +222,69 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
       writer.literal("INT64");
       writer.endFunCall(castFrame);
       break;
+    case REGEXP_SUBSTR:
+      unparseRegexSubstr(writer, call, leftPrec, rightPrec);
+      break;
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  private void unparseRegexSubstr(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    final SqlWriter.Frame regexFrame;
+    switch (call.operandCount()) {
+    case 2:
+      regexFrame = writer.startFunCall("REGEXP_EXTRACT");
+      for (SqlNode operand : call.getOperandList()) {
+        writer.sep(",");
+        operand.unparse(writer, leftPrec, rightPrec);
+      }
+      writer.endFunCall(regexFrame);
+      break;
+    case 3:
+      regexFrame = writer.startFunCall("REGEXP_EXTRACT");
+      unparseSubstr(writer, call, leftPrec, rightPrec);
+      writer.sep(",");
+      call.operand(1).unparse(writer, leftPrec, rightPrec);
+      writer.endFunCall(regexFrame);
+      break;
+    case 4:
+      regexFrame = writer.startFunCall("REGEXP_EXTRACT_ALL");
+      unparseSubstr(writer, call, leftPrec, rightPrec);
+      writer.sep(",");
+      call.operand(1).unparse(writer, leftPrec, rightPrec);
+      writer.endFunCall(regexFrame);
+      unparseOccurrenceNode(writer, call);
+      break;
+    case 5:
+      regexFrame = writer.startFunCall("REGEXP_EXTRACT_ALL");
+      unparseSubstr(writer, call, leftPrec, rightPrec);
+      writer.sep(",");
+      if (call.operand(4).toString().equals("'i'")) {
+        writer.literal("'(?i)".concat(call.operand(1).toString()));
+      } else {
```

Comment:
there are I guess 5 options and not just 2, so every other case will not go to else. Please check the Teradata document once

Reply:
I am aware about that. There are total 6 valid arguments and some arguments can be combined increasing unique valid arguments. We need to discuss if we are going to handle all of them.